DigitalExcellence/dex-backend
# XUnit integration tests

## Description

This PR adds XUnit integrations tests. The XUnit tests are runned from within a Docker container. You will need to make sure to have the backend being runned via docker-compose.
This PR also removed Postman tests that where causing the pipeline to fail, tests removed where added back to the new XUnit integration test project.

## Type of change


## Checklist


## Steps to Test or Reproduce

Test that the tests are runned on your local machine via Docker.

1. Pull the latest changes from this branch
2. execute `docker-compose up --build` . Make sure to use the` -- build` flag because otherwise you will use a older build which will not work for this PR.
3. Build the XUnit image via  `docker build -t dex-backend_xunit -f API.Tests/Dockerfile .`
3. When the backend runs execute the XUnit container via the following command. Make sure to be in the root of the repository and not in any sub folders etc. `docker run -e "ApiAddress=https://api:5001/" -e "IdentityAddress=http://identity:5004/" --network  dex-backend_mssql-network dex-backend_xunit`
4. All tests should pass


alternatively you can run XUnit without docker and then you need to make sure that your local machines has the two following system variables:

- ApiAddress=https://localhost:5001/
- IdentityAddress=http://localhost:5005/
Now you can run the integration tests via the Test Explorer in Visual studio or any other IDE.

## Link to issue

Closes: #364
